src/pages/AddProposal/AddProposal.component.tsx

```diff
@@ -45,16 +44,16 @@ export const AddProposal = ({
     name: AddProposalInputs.Values,
   });
 
-  const dispatch = useDispatch();
-
   const govOption = useWatch({
     name: AddProposalInputs.SendProposalContract,
     control,
   });
 
   useEffect(() => {
-    dispatch(proposalsActions.setGovernor(govOption));
-  }, [govOption, dispatch]);
+    if (onGovernorChange) {
```

Comment:
This check could be removed once you make the `onGovernorChange` required
```suggestion
    onGovernorChange(govOption);
```